### PR TITLE
add support for proxmark3 json

### PR DESCRIPTION
**Pull request by GameTec-live**

### CODE REVIEW

File: chameleonultragui/lib/helpers/mifare_classic/general.dart [hunk 1]
```diff
@@ -217,3 +231,25 @@ List<Uint8List> mfClassicGetKeysFromDump(List<Uint8List> dump) {
 
   return keys;
 }
+
+TagType mfClassicTypeToTagType(MifareClassicType type) {
```
Comment (Foxushka) [inline chameleonultragui/lib/helpers/mifare_classic/general.dart:235]: mfClassicGetChameleonTagType exists

File: chameleonultragui/lib/helpers/mifare_classic/general.dart [hunk 2]
```diff
@@ -217,3 +231,25 @@ List<Uint8List> mfClassicGetKeysFromDump(List<Uint8List> dump) {
...
+    MifareClassicType.m1k: TagType.mifare1K,
+    MifareClassicType.m2k: TagType.mifare2K,
+    MifareClassicType.m4k: TagType.mifare4K,
+    MifareClassicType.mini: TagType.mifareMini,
+  };
+
+  return typeMap[type] ?? TagType.unknown;
+}
+
+MifareClassicType tagTypeToMfClassicType(TagType type) {
```
Comment (Foxushka) [inline chameleonultragui/lib/helpers/mifare_classic/general.dart:246]: chameleonTagTypeGetMfClassicType exists

File: chameleonultragui/lib/sharedprefsprovider.dart [hunk 3]
```diff
@@ -78,22 +78,67 @@ class CardSave {
...
+    final List<int> atqa;
+    final List<int> ats;
+    final String name;
+    final TagType tag;
+    final Color color;
     List<Uint8List> tagData = [];
-    for (var block in encodedData) {
-      tagData.add(Uint8List.fromList(List<int>.from(block)));
+
+    if (data['Created'] == "proxmark3") {
```
Comment (Foxushka) [inline chameleonultragui/lib/sharedprefsprovider.dart:92]: big no, this is cardsave, intended only for OUR card saves. Implement converter somewhere outside

**Review by Foxushka — CHANGES_REQUESTED** [review]

File: chameleonultragui/lib/sharedprefsprovider.dart (hunk 3 above)
Comment (GameTec-live) [reply chameleonultragui/lib/sharedprefsprovider.dart:92]: pm3 json is json, so the function fromjson should handle whatever json you throw at it. And adding a check like that in the import function would require looking at the json and then we try to unnescesarely parse json

**Review by Foxushka — APPROVED** [review]

**Pull request merged**